R/AssayLinks.R

```diff
@@ -257,6 +272,52 @@ setMethod("[", c("AssayLinks", "list"),
               hits = hits)
 }
 
+
+## The function takes a Features object, the assay name*s* to link from, the 
+## assay name to link to and the feature variable names contained in the rowData 
+## of the assay*s* to link from and the assay to link to. The function returns 
+## an AssayLink object that links the parent assay*s* to the child assay given 
+## the relationship between the corresponding feature variables.
+.create_assay_link_multi <- function(object, 
+                                     from, 
+                                     to, 
+                                     varsFrom, 
+                                     varTo) {
+    if (missing(varsFrom) | missing(varTo)) {
+        ## Create the list of hits between the child assay and the parent assays
+        ## based on the rownames 
+        rdTo <- cbind(rowData(object[[to]]), RowNames = rownames(object[[to]]))
+        hitsList <- lapply(seq_along(from), function(ii) {
+            rdFrom <- cbind(rowData(object[[from[ii]]]), 
+                            RowNames = rownames(object[[from[[ii]]]]))
+            .create_assay_link(rdFrom = rdFrom,
+                               rdTo = rdTo,
+                               from[ii], to,
+                               "RowNames", "RowNames")@hits
+        })
+        varsFrom <- "RowNames"
```

Comment:
This conditional code chunk might seem dirty but is necessary for matching `rownames` to parent assays (only possible when called within `joinAssays`). Maybe you can think about something more clever?